src/machine.rs

```diff
@@ -1093,6 +1093,30 @@ impl OlmMachine {
             stream.for_each(move |item| send_user_identities_to_callback(callback_ref, item)).await;
         });
     }
+    /// Register a callback which will be called whenever a secret is received.
+    ///
+    /// `callback` should be a function that takes 2 arguments the secret name
+    /// and value (base64) and returns a Promise.
+    #[wasm_bindgen(js_name = "registerReceiveSecretCallback")]
+    pub async fn register_secret_receive_callback(&self, callback: Function) {
+        let stream = self.inner.store().secrets_stream();
+        let store = self.inner.store().clone();
+        //fire up a promise chain which will call `cb` on each result from the stream
+        spawn_local(async move {
+            let callback_ref = &callback;
```

Comment:
Yes me too, but the test fails if it is not used with "Did you forget to wait for something async in your test?" 
I don't understand why though

Reply:
which test?

Reply:
I don't know exactly what test, here is a failing run https://github.com/matrix-org/matrix-rust-sdk-crypto-wasm/actions/runs/6418397170/job/17426095599

But looks unrelated, it might just be flacky?

Reply:
I would imagine it is a flaky test, yes